x/records/keeper/epoch_unbonding_record.go

```diff
@@ -143,3 +143,40 @@ func (k Keeper) IterateEpochUnbondingRecords(ctx sdk.Context,
 		i++
 	}
 }
+
+// GetEpochUnbondingRecordByEpoch returns a epochUnbondingRecord from its epochNumber
+func (k Keeper) GetHostZoneUnbondingByChainId(ctx sdk.Context, id uint64, chainId string) (val *types.HostZoneUnbonding, found bool) {
```

Comment:
This "list as a dict" paradigm is nice. Wonder if we should build it out across the codebase. We have other lists we'd prefer to interact with "as maps" but we iterate them across the codebase in a haphazard manner, rather than using a helper like this.

Reply:
Yeah, might make sense to do so! It is kind of inefficient, but for lists where we know we won't have many elements it can work